cram/cram_encode.c

```diff
@@ -1921,6 +1921,12 @@ int cram_encode_container(cram_fd *fd, cram_container *c) {
                 // Do not confuse with fd->ref_free which is a pointer to a
                 // reference string to free.
                 c->ref_free = 1;
+            } else {
+                // Double check for broken input.  We shouldn't have
+                // embedded references enabled for unmapped data, but our
+                // data could be broken.
+                embed_ref = 0;
+                no_ref = 1;
```

Comment:
Should this also set `c->no_ref` [as done elsewhere](https://github.com/samtools/htslib/blob/b3a4c0a62240faf437105d7604897cf7046e3474/cram/cram_encode.c#L1948)?  `no_ref` is reset from `c->no_ref` later.  `c->no_ref` is also used in [`cram_encode_compression_header()`](https://github.com/samtools/htslib/blob/b3a4c0a62240faf437105d7604897cf7046e3474/cram/cram_encode.c#L90), which is called from this function.

Reply:
I was wary of changing anything outside of the immediate code it was calling due to potential threading issues, but the container is OK as it's never accessed by more than one thread once we're at the encode stage.  It looks safe to do this.  I'm not sure if it matters though for us, but there are comments about using no_ref as a flag during decode to avoid java implementation validation errors.